lib/commands/implementations/breaking-news.js

```diff
@@ -0,0 +1,59 @@
+const moment = require('moment');
+const _ = require('lodash');
+
+const Command = require('../command-interface');
+const CommandOutput = require('../command-output');
+const { makeMute } = require('../../chat-utils/punishment-helpers');
+const parseDurationToSeconds = require('../../chat-utils/duration-parser');
+const formatDuration = require('../../chat-utils/format-duration');
+
+function breakingNews(defaultMaxAge) {
+  let enabled = false;
+
+  return (input, services, rawMessage) => {
+    const matched = /(\d+[HMDShmds])?/.exec(input);
+    const maxAgeStr = _.get(matched, 1, defaultMaxAge).toLowerCase();
+
+    enabled = !enabled;
+
+    if (!enabled) {
+      services.messageRelay.stopRelay('breakingnews');
+      return new CommandOutput(null, 'Breaking News mode turned off.');
+    }
+
+    const maxAge = parseDurationToSeconds(maxAgeStr);
+    const listener = services.messageRelay.startListenerForChatMessages('breakingnews');
+    listener.on('message', (data) => {
+      const message = data.message.trim().toLowerCase();
+      if (!services.messageMatching.hasTwitterLink(message)) return;
+      services.twitterApi
+        .getTweetDatetimeFromMessage(message)
```

Comment:
Consider dumping the tweet id into a Map or something along with the post timestamp to cache it. I would think there is a high likelihood of multiple people posting the same tweet causing unnecessary api calls.